suzieq/cli/sqcmds/context_commands.py

```diff
@@ -28,15 +28,32 @@
     "datadir",
     description="Set the data directory for the command"
 )
+@argument(
+    'rest_server_ip', description='IP address of the REST server'
+)
+@argument(
+    'rest_server_port', description='Port of the REST server'
+)
+@argument(
+    'rest_api_key', description='API key for the REST server'
+)
+@argument(
+    'rest_use_https', description='Use HTTPS for the REST server',
+    choices=['True', 'False']
+)
 def set_ctxt(
         pager: str = "",
-        hostname: typing.List[str] = [],
+        hostname: typing.List[str] = None,
         start_time: str = "",
         end_time: str = "",
-        namespace: typing.List[str] = [],
+        namespace: typing.List[str] = None,
         engine: str = "",
         datadir: str = "",
         col_width: int = 50,
+        rest_server_ip: str = "",
+        rest_server_port: str = "",
+        rest_api_key: str = "",
+        rest_use_https: str = ""
 ):
     """set certain contexts for subsequent commands. Cmd is additive"""
     plugin_ctx = context.get_context()
```

Comment:
You need to set the parameter in plugin_ctx.ctxt not in the NubiaSuzieqContext object

Reply:
Fixed